tests/d3d12_dxvk_interop_device.c

```diff
@@ -0,0 +1,157 @@
+    ID3D12Device *device;
+    VkInstance vkInstance;
+    unsigned int value;
+    VkDevice vkDevice;
+    VkBuffer vkBuffer;
+    VkFormat format;
+    HRESULT hr;
+
+    static const unsigned int data_values[] = {0xdeadbeef, 0xf00baa, 0xdeadbeef, 0xf00baa};
+    static const uint32_t update_data[3] = { 0x1020304, 0xc0d0e0f, 0x5060708};
+
+    // Initialize pointer to GDPA function
+    if (!init_vulkan_loader())
+        return;
+
+    if (!init_test_context(&context, NULL))
+        return;
+
+    device = context.device;
+
+    if (FAILED(hr = ID3D12Device_QueryInterface(device,
+            &IID_ID3D12DXVKInteropDevice1, (void **)&interop_device)))
+    {
+        skip("ID3D12DXVKInteropDevice1 not implemented.\n");
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(hr = ID3D12DXVKInteropDevice1_GetVulkanHandles(interop_device,
+        &vkInstance,
+        &vkPhysicalDevice,
+        &vkDevice)))
+    {
+        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_GetVulkanHandles failed %#x.\n", hr);
+        destroy_test_context(&context);
+        return;
+    }
+
+    pfn_vkCmdUpdateBuffer = (PFN_vkCmdUpdateBuffer)pfn_vkGetDeviceProcAddr(vkDevice, "vkCmdUpdateBuffer");
+    if (!pfn_vkCmdUpdateBuffer)
+    {
+        ok(pfn_vkCmdUpdateBuffer, "vkCmdUpdateBuffer not found.\n");
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(hr = ID3D12DXVKInteropDevice1_CreateInteropCommandAllocator(interop_device,
+        D3D12_COMMAND_LIST_TYPE_DIRECT,
+        VK_QUEUE_FAMILY_IGNORED,
+        &allocator)))
+    {
+        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_CreateInteropCommandAllocator failed %#x.\n", hr);
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(hr = ID3D12Device_CreateCommandList(device, 0, D3D12_COMMAND_LIST_TYPE_DIRECT,
+            allocator, NULL, &IID_ID3D12GraphicsCommandList, (void **)&command_list)))
+    {
+        ok(hr == S_OK, "ID3D12Device_CreateCommandList failed %#x.\n", hr);
+        destroy_test_context(&context);
+        return;
+    }
+
+    if (FAILED(ID3D12DXVKInteropDevice1_BeginVkCommandBufferInterop(interop_device,
+        (ID3D12CommandList*)command_list,
+        &vkCmdBuf)))
+    {
+        ok(hr == S_OK, "ID3D12DXVKInteropDevice1_GetVulkanCmdHandles failed %#x.\n", hr);
+        destroy_test_context(&context);
+        return;
+    }
+
+    queue = context.queue;
+
+    // Create a D3D12 Buffer
+    buffer = create_default_buffer(device,
+                sizeof(data_values),
+                D3D12_RESOURCE_FLAG_NONE,
+                D3D12_RESOURCE_STATE_COPY_DEST);
+    upload_buffer_data(buffer, 0, sizeof(data_values), data_values, queue, command_list);
+    reset_command_list(command_list, allocator);
+
+
+    // Export it to VK and use VK functions to manipulate it
+    ID3D12DXVKInteropDevice1_GetVulkanResourceInfo1(interop_device,
+        buffer,
+        (uint64_t*)&vkBuffer,
+        &offset,
```

Comment:
This needs to be UINT64 to get the 8 byte alignment on 32-bit. Both the handle and offset it seems.